fe/fe-core/src/main/java/com/starrocks/qe/ShortCircuitExecutor.java

```diff
@@ -63,6 +67,9 @@ protected ShortCircuitExecutor(ConnectContext context, PlanFragment planFragment
         this.isBinaryRow = isBinaryRow;
         this.enableProfile = enableProfile;
         this.protocol = protocol;
+        if (enableProfile) {
+            this.perBeExecutionProfile = new HashMap<>();
```

Comment:
Maybe init it with `Collections.emptyMap()` if `!enableProfile` to give it a safer default behavior.